ssh2/session.pyx

```diff
@@ -274,12 +275,15 @@ cdef class Session:
         """
         cdef int rc
         cdef bytes b_username = to_bytes(username)
-        cdef bytes b_password = to_bytes(password)
         cdef const char *_username = b_username
-        cdef const char *_password = b_password
-        with nogil:
-            rc = c_ssh2.libssh2_userauth_keyboard_interactive(
-                self._session, _username, _password)
+
+        def passwd():
+            return(password)
```

Comment:
As far as tests, I don't know how this feature originally got past the tests because you can't call that C function with a string as the password and as such the existing test for this is fine. I'm not sure how you'd even test for this other than feeding the function another function instead of a string now.

Reply:
Can't release GIL either.

Reply:
The original function was a contribution lacking tests - it should not have been merged without.

There is no existing test for `userauth_keyboardinteractive`, it needs one written.

Reply:
Was missing a pyd header declaration change because the original patch declares the `libssh2_userauth_keyboard_interactive` function incorrectly.

Reply:
There is already tests for this function, https://github.com/ParallelSSH/ssh2-python/blob/master/tests/test_session.py#L30-L34